### PR TITLE
feat: cancel model download

### PR DESCRIPTION
**Pull request by anuraagdjain**
## Pull Request Checklist

- [x] **Description:** Briefly describe the changes in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation?
- [ ] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?

---

## Description
[Insert a brief description of the changes made in this pull request]

- Ability to cancel the download of a model. Addresses https://github.com/open-webui/open-webui/issues/526
- WIP: Deleting the model from Ollama

image

---

### Added

- [List any new features or additions]
- Show **X** button next to the download progress, allowing user to cancel the download.

### Fixed

- [List any fixes or corrections]

### Changed

- [List any changes or updates]

### Removed

- [List any removed features or files]

### CODE REVIEW

File: src/lib/components/chat/Settings/Models.svelte [hunk 1]
```diff
@@ -599,12 +613,25 @@
...
 									<div class="">
+										<div class="flex flex-row space-x-4">
 										<div
 											class="dark:bg-gray-600 bg-gray-500 text-xs font-medium text-gray-100 text-center p-0.5 leading-none rounded-full"
 											style="width: {Math.max(15, modelDownloadStatus[model].pullProgress ?? 0)}%"
 										>
 											{modelDownloadStatus[model].pullProgress ?? 0}%
 										</div>
+										<button
+										class="px-2.5 bg-gray-100 hover:bg-gray-200 text-gray-800 dark:bg-gray-850 dark:hover:bg-gray-800 dark:text-gray-100 rounded-lg transition"
```
Comment (anuraagdjain) [inline src/lib/components/chat/Settings/Models.svelte:622]: Q: If the UI of **X** needs to be changed, please tell me the class as I am still trying to figure out.

File: src/lib/components/chat/Settings/Models.svelte [hunk 2]
```diff
@@ -466,6 +467,17 @@
 		ollamaVersion = await getOllamaVersion(localStorage.token).catch((error) => false);
 		liteLLMModelInfo = await getLiteLLMModelInfo(localStorage.token);
 	});
+	const deleteModelPull = async(model: string) => {
+		const {reader} = modelDownloadStatus[model];
+		if(reader){
+			await reader.cancel();
+			delete modelDownloadStatus[model];
+			await deleteModel(localStorage.token, model);
```
Comment (anuraagdjain) [inline src/lib/components/chat/Settings/Models.svelte:475]: Calling delete endpoint with `mistral` fails with 

```json
{
    "detail": "Model 'mistral' was not found"
}
```

What if a tag is not provided, should I assume `:latest` and just append that to the `model` ? 

For example: 
- if the input is `mistral` then add `:latest` => `mistral:latest` 
- if the input is `mistral:7b` then just send the data

**Comment by tjbck** [comment]
It should actually cancel the model download from the backend now, let me know if I broke anything!

File: src/lib/components/chat/Settings/Models.svelte (hunk 1 above)
Comment (tjbck) [reply src/lib/components/chat/Settings/Models.svelte]: Updated!

File: src/lib/components/chat/Settings/Models.svelte (hunk 2 above)
Comment (tjbck) [reply src/lib/components/chat/Settings/Models.svelte:503]: I'm not sure if Ollama provides APIs for deleting partially downloaded blobs, we might want to look into that!

Comment (tjbck) [reply src/lib/components/chat/Settings/Models.svelte:511]: I'll merge this for now, but let's come back to this later. Thanks!

**Comment by anuraagdjain** [comment]
Thanks, I"ll check this on master this week and open a followup PR if needed.

**Pull request merged**